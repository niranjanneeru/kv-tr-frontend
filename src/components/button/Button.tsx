import { FC } from 'react';
import './style.css';

type InputPropType = {
    label: string;
    buttonType: 'submit' | 'reset' | 'button';
    onClickCallback: (e) => void;
    buttonDesign?: string;
}

const Button: FC<InputPropType> = ({ label, buttonType, onClickCallback, buttonDesign = "primary" }) => {
    const buttonClass = getCSSClass(buttonDesign);

    return (
        <button className={buttonClass} type={buttonType} onClick={onClickCallback}>{label}</button>
    );
};

export default Button;

function getCSSClass(buttonDesign: string) {
    if (buttonDesign === "primary") return "login-button";
    if (buttonDesign === "cancel") return "cancel-button";
    if (buttonDesign === "create") return "primary";
}
