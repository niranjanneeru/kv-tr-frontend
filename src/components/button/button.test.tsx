import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';
import '@testing-library/jest-dom';

describe('Test for Button Component', () => {
	it('If Label is correct', () => {
		render(<Button label='create' buttonType='submit' onClickCallback={() => {}} />);
		const dom = screen.getByTestId('button-create-id');

		expect(dom).toHaveTextContent('create');
	});

	it('If Type is correct', () => {
		render(<Button label='create' buttonType='submit' onClickCallback={() => {}} />);
		const dom = screen.getByTestId('button-create-id');

		expect(dom).toHaveAttribute('type', 'submit');
		expect(dom).not.toHaveAttribute('type', 'cancel');
	});

	it('To Match Snapshot', () => {
		render(<Button label='create' buttonType='submit' onClickCallback={() => {}} />);
		const dom = screen.getByTestId('button-create-id');

		expect(dom).toMatchSnapshot();
	});

	it('If Callback works', () => {
		const cb = jest.fn();

		render(<Button label='create' buttonType='submit' onClickCallback={cb} />);
		const dom = screen.getByTestId('button-create-id');

		fireEvent.click(dom);

		// dom.click();

		expect(cb).toHaveBeenCalled();
		expect(cb).not.toBeCalledTimes(2);
	});
});
