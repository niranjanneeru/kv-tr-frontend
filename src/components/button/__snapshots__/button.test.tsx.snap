// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Button Component To Match Snapshot 1`] = `
<button
  class="login-button"
  data-testid="button-create-id"
  type="submit"
>
  create
</button>
`;
